bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/PageImpl.java

```diff
@@ -0,0 +1,451 @@
+     * @param request request to be wrapped
+     * @param page page to be referenced as statically containing the current page content
+     * @return {@link SlingHttpServletRequest} a {@link SlingHttpServletRequestWrapper} containing given page and request
+     */
+    protected static SlingHttpServletRequest getHierarchyServletRequest(@Nonnull SlingHttpServletRequest request,
+                                                                        @Nonnull com.day.cq.wcm.api.Page page,
+                                                                        @Nullable com.day.cq.wcm.api.Page entryPage) {
+        // Request attribute key of the component context
+        final String ATTR_COMPONENT_CONTEXT = "com.day.cq.wcm.componentcontext";
+
+        // Request attribute key of the current page
+        final String ATTR_CURRENT_PAGE = "currentPage";
+
+        SlingHttpServletRequest wrapper = new SlingHttpServletRequestWrapper(request);
+
+        ComponentContext componentContext = (ComponentContext) request.getAttribute(ATTR_COMPONENT_CONTEXT);
+
+        // When traversing child pages the currentPage must be updated
+        wrapper.setAttribute(ATTR_COMPONENT_CONTEXT, new HierarchyComponentContextWrapper(componentContext, page));
+        wrapper.setAttribute(ATTR_CURRENT_PAGE, page);
+        wrapper.setAttribute(ATTR_HIERARCHY_ENTRY_POINT_PAGE, entryPage);
+
+        return wrapper;
+    }
+
+
+    // style helpers zone
+
+    /**
+     * Returns the first numeric selector. The default value is 0
+     *
+     * @return {@link int} the defined traversal depth or 0 if none defined
+     */
+    protected static int getPageTreeTraversalDepth(Style style) {
+        // Depth of the tree of pages
+        final String PN_STRUCTURE_DEPTH = "structureDepth";
+
+        Integer pageTreeTraversalDepth = null;
+
+        if (style != null) {
+            pageTreeTraversalDepth = style.get(PN_STRUCTURE_DEPTH, Integer.class);
+        }
+
+        if (pageTreeTraversalDepth == null) {
+            return 0;
+        }
+
+        return pageTreeTraversalDepth;
+    }
+
+    /**
+     * Get request's entry point attribute value
+     */
+    protected static com.day.cq.wcm.api.Page requestGetHierarchyEntryPoint(@Nonnull SlingHttpServletRequest request) {
+        return (com.day.cq.wcm.api.Page) request.getAttribute(ATTR_HIERARCHY_ENTRY_POINT_PAGE);
+    }
+
+    /**
+     * Returns the page structure patterns to filter the child pages to be exported.
+     * The patterns can either be stored on the template policy of the page or provided as a request parameter
+     *
+     * @param request request
+     * @return {@link List} list of page structure patterns
+     */
+    @Nonnull
+    protected static List<Pattern> getStructurePatterns(@Nonnull SlingHttpServletRequest request, Style currentStyle) {
+        // List of Regexp patterns to filter the exported tree of pages
+        final String PN_STRUCTURE_PATTERNS = "structurePatterns";
```

Comment:
same as above: 
This constant should be defined in the Page API. IMHO there should be an API method to retrieve it.